gallery/src/styles/NxFormLayout/NxFormLayoutExample.tsx

```diff
@@ -4,61 +4,112 @@
  * the terms of the Eclipse Public License 2.0 which accompanies this
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React from 'react';
+import React, { useState } from 'react';
 
 import { NxCheckbox } from '@sonatype/react-shared-components';
 import { NxRadio } from '@sonatype/react-shared-components';
+import { NxStatefulTextInput } from '@sonatype/react-shared-components';
+import { NxButton } from '@sonatype/react-shared-components';
 
 export default function NxFormLayoutExample() {
+  function validator(val: string) {
+    return val.length ? null : 'Must be non-empty';
+  }
+
+  const [isRed, setIsRed] = useState(false),
+      [isBlue, setIsBlue] = useState(false),
+      [isGreen, setIsGreen] = useState(false),
+      [isDisabled, setIsDisabled] = useState(false),
+      toggleRed = () => setIsRed(!isRed),
+      toggleBlue = () => setIsBlue(!isBlue),
+      toggleGreen = () => setIsGreen(!isGreen),
+      toggleDisabled = () => setIsDisabled(!isDisabled);
+
+  const [color, setColor] = useState<string | null>(null);
+
   return (
     <form className="nx-form">
       <div className="nx-form-group">
         <label className="nx-label">
           <span className="nx-label__text">Label</span>
-          <input type="text" className="nx-text-input"/>
+          <NxStatefulTextInput/>
         </label>
       </div>
       <div className="nx-form-group">
         <label className="nx-label nx-label--optional">
           <span className="nx-label__text">Label</span>
-          <input type="text" className="nx-text-input"/>
+          <NxStatefulTextInput validator={validator}/>
         </label>
       </div>
       <div className="nx-form-group">
         <label className="nx-label nx-label--optional">
           <span className="nx-label__text">Long field</span>
           <span className="nx-sub-label">This is a sub-label. The field element below is wider than the default.</span>
-          <input type="text" className="nx-text-input nx-text-input--long"/>
+          <NxStatefulTextInput className="nx-text-input--long"/>
         </label>
       </div>
       <fieldset className="nx-fieldset">
-        <legend className="nx-label">Checkboxes</legend>
-        <NxCheckbox isChecked={false}>Checkbox 1</NxCheckbox>
-        <NxCheckbox isChecked={true}>Checkbox 2</NxCheckbox>
-        <NxCheckbox isChecked={false}>Checkbox 3</NxCheckbox>
+        <legend className="nx-legend">
+          <span className="nx-legend__text">
+            Checkboxes ({isRed && ' Red'}{isBlue && ' Blue'}{isGreen && ' Green'} )
+          </span>
+        </legend>
+        <NxCheckbox checkboxId="subscribe-check" onChange={toggleRed} isChecked={isRed}>
+          Red
+        </NxCheckbox>
+        <NxCheckbox checkboxId="no-label-check" onChange={toggleBlue} isChecked={isBlue}>Blue</NxCheckbox>
```

Comment:
This id doesn't make sense.  Overall I'd suggest simplifying the VDOM of this part of the example.  We already demonstrate all the edge-case capabilities on the `NxCheckbox` page itself, no need to have anything more than a straightforward set of NxCheckboxes (and NxRadios) on this page.

Reply:
> This id doesn't make sense. Overall I'd suggest simplifying the VDOM of this part of the example. We already demonstrate all the edge-case capabilities on the `NxCheckbox` page itself, no need to have anything more than a straightforward set of NxCheckboxes (and NxRadios) on this page.

Cleaned up and consolidated here: https://github.com/sonatype/sonatype-react-shared-components/pull/122/commits/bf7e0b1dc09a3fba78513c1d932c38a63fe9b941